### PR TITLE
#4347 [Manager] Use redux to refact the worker, broker and zookee…

### PR DESCRIPTION
**Pull request by eechih**
…per context

### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4347 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/store/apiWrapper.js [hunk 1]
```diff
@@ -77,8 +92,200 @@ export const wrapUpdatePipeline = values =>
...
+    brokerApi.remove(params).then(res => {
       if (res.errors) reject(res);
       else resolve(params);
     }),
   );
+
+export const wrapFetchBrokers = () =>
+  new Promise((resolve, reject) =>
+    brokerApi.getAll({ group: brokerGroup }).then(res => {
+      if (res.errors) reject(res);
```
Comment (saivirtue) [inline ohara-manager/client/src/store/apiWrapper.js]: I think we don't have errors object now :)
you can use `.then().catch()` instead.

File: ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+const waitBroker$ = action =>
+  timer(0, 1000).pipe(
+    switchMap(() =>
+      from(brokerApi.get(action.payload?.brokerSettings)).pipe(
+        map(res => {
+          console.log(res);
+          return res.data;
+        }),
+        filter(data => data.state === 'RUNNING'),
```
[the remark this answers is not in the document]
Comment (eechih) [reply ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: > We have a state enum SERVICE_STATE in clusterInterface.ts

Got it.

> And we should consider failed case

Thanks, I will think about it. :)

File: ohara-manager/client/src/store/apiWrapper.js (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/src/store/apiWrapper.js]: apiWrapper 即將廢棄～

**Comment by eechih** [comment]
@saivirtue Have any other comments?

@konekoya @wu87988622 Please take a look this PR when you have time :)

**Comment by saivirtue** [comment]
please retry e2e tests :)

**Comment by eechih** [comment]
retry manager-e2e

**Comment by eechih** [comment]
retry manager-e2e

**Comment by eechih** [comment]
retry manager-e2e

**Pull request merged**